### ISSUE TITLE
Negative token expiration when changing client session max lifetime

### ISSUE THREAD

**Issue by sschu** [keycloak/keycloak#24936]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

oidc

### Describe the bug

When adding a client session max limit, Keycloak will generate refresh and access tokens with negative token lifetime.

### Version

22.0.5, nightly

### Expected behavior

Keycloak creates a new client session and a new token with positive token expiration based on the client session settings.

### Actual behavior

Keycloak uses the original start date of the first client session and calculates the expiry date based on the now reduced client session max.

### How to Reproduce?

1. Do an authcode flow to retrieve an access+refresh token.
2. Enter a client session max of 1 minute in the realm settings.
3. Wait at least one minute.
4. Do another authcode flow to retrieve an access+refresh token, both tokens will have expiration date in the past and expires_in is negative in the response.

### Anything else?

This is really a corner case, probably not that critical.

**Comment by sschu** [comment]
I can still reproduce this on Keycloak 23.0.6 (see screenshot 
<img width="462" alt="Screenshot 2024-02-15 at 18 09 01" src="https://github.com/keycloak/keycloak/assets/3833769/b9dea158-7ebe-4a5a-9c8a-5d76f3c9027c">
). Its easy to reproduce: login to the admin console, change the client session max to 1 minute, wait for 1 minute for access token to expire, click anywhere (the console will get a new token using refresh flow), observe the negative expiration.

**Comment by sschu** [comment]
Instead of clicking to trigger a refresh flow, you can also reload the page. The problem also occurs there and sends the admin console in a nasty reload loop:
<img width="1643" alt="Screenshot 2024-02-15 at 18 20 36" src="https://github.com/keycloak/keycloak/assets/3833769/b6ce435c-8829-4e81-9249-621cbb1adaae">

**Comment by douglaspalmer** [comment]
@sschu I can reproduce it manually; I'll try to fix the test.

**Comment by mposolda** [comment]
@sschu This might not be that trivial to fix and at the same time, it is corner case. We don't prioritize this ATM within Keycloak team. But PR is welcome.

**Comment by sschu** [comment]
@mposolda I agree with the priority. I am not even sure what the expected behaviour here is. I would say shortening the session timespans should have a retroactive effect. This means Keycloak should just act as if the session is expired. This might also make it easier to fix this (just respond with an error if the expiration would be negative).
